_posts/2018-12-29-dask-on-hpc.md

```diff
@@ -0,0 +1,191 @@
+
+# Hadoop/Spark clusters
+from dask_yarn import YARNCluster
+cluster = YarnCluster(environment='environment.tar.gz', worker_vcores=2)
+
+# Cloud/Kubernetes clusters
+from dask_kubernetes import KubeCluster
+cluster = KubeCluster(pod_spec={...})
+```
+
+Dask is more than just a tool to us; it is a gateway to thinking about a completely different way of providing computing infrastructure to our users. Dask opens up the door to cloud computing technologies (such as elastic scaling and object storage) and makes us rethink what an HPC center should really look like.
+
+
+### 6.  Cost and Collaboration
+
+Dask is free and open source, which means we do not have to rebalance our budget and staff to address the new immediate need of data analysis tools.
+We don't have to pay for licenses, and we have the ability to make changes to the code when necessary.  The HPC community has good representation among Dask developers.  It's easy for us to participate and our concerns are well understood.
+
+<!-- WR: Mention quick response to new use cases / demands as another benefit of the collaborative
+approach?  And hint towards dask-mpi here? -->
+
+
+# What needs work
+
+### 1.  Heterogeneous resources handling
+
+Often we want to include different kinds of HPC nodes in the same deployment.
+This includes situations like the following:
+
+-  Workers with low or high memory,
+-  Workers with GPUs,
+-  Workers from different node pools.
+
+Dask provides some support for this heterogeneity already, but not enough.
+We see two major opportunities for improvement.
+
+-  Tools like Dask-Jobqueue should make it easier to manage multiple worker
+   pools within the same cluster.  Currently the deployment solution assumes
+   homogeneity.
+-  It should be easier for users to specify which parts of a computation
+   require different hardware.  The solution today works, but requires more
+   detail from the user than is ideal.
+
+
+### 2.  Coarse-Grained Diagnostics and History
+
+Dask provides a number of profiling tools that deliver real-time diagnostics at the individual task-level, but there is no way today to analyze or profile your Dask application at a coarse-grained level, and no built-in way to track performance over long periods of time.
+
+Having more tools to analyze bulk performance would be helpful when making design decisions and future architecture choices.
+
+Having the ability to persist or store history of computations (`compute()` calls)
+and tasks executed on a scheduler could be really helpful to track problems and potential performance improvements.
+
+<!-- JJH: One tangible idea here would be a benchmarking suite that helps users make decisions about how to use dask most effectively.
+ -->
+
+### 3.  Scheduler Performance on Large Graphs
+
+HPC users want to analyze Petabyte datasets on clusters of thousands of large nodes.
+
+While Dask can theoretically handle this scale, it does tend to slow down a bit,
```

Comment:
`slow down a bit` 
I would consider this is a strong understatement given my experience. 
In several occasions, I found myself into situations where the scheduler cannot handle the load and I had to rewrite code in order to account for this (by manually considering subsets of my datasets).

Reply:
Did you have memory problem on workers, or really it as the scheduler not responding?

I agree that this can be more than a slow down as seen per a user perspective. I've not pushed Dask to its task number limit since several month though, maybe it has improved.